src/com/wsscode/pathom/core.cljc

```diff
@@ -828,7 +828,7 @@
   (fn wrap-parser-exception-internal [env tx]
     (let [errors (atom {})]
       (let-chan [res (parser (assoc env ::errors* errors) tx)]
-        (cond-> res
+        (cond-> (if (keyword? res) {} res)
```

Comment:
Maybe it would be more self-documenting if we explicitly checked for `::reader-error`, we don't expect other kws for now.